src/ruby/yast/rspec/scr.rb

```diff
@@ -0,0 +1,84 @@
+require "rspec"
+require "yast"
+
+# RSpec extension to handle several agent operations.
+#
+# @example usage
+#
+# require 'yast/rspec/scr'
+#
+# RSpec.configure do |c|
+#   c.include Yast::RSpec::SCR
+# end
+#
+# describe YaST::SCR do
+#   before do
+#     chroot = File.join(File.dirname(__FILE__), "test_chroot")
+#     change_scr_root(chroot)
+#   end
+#
+#   after do
+#     reset_scr_root
+#   end
+#
+#   describe "#Read" do
+#     it "works with the .proc.meminfo path"
+#       # This uses the #path helper from SCRStub and
+#       # reads from ./test_chroot/proc/meminfo
+#       values = Yast::SCR.Read(path(".proc.meminfo"))
+#       expect(values).to include("key" => "value")
+#     end
+#   end
+# end
+module Yast
+  module RSpec
+    module SCR
+      # Shortcut for generating Yast::Path objects
+      #
+      # @param route [String] textual representation of the path
+      # @return [Yast::Path] the corresponding Path object
+      def path(route)
+        Yast::Path.new(route)
+      end
+
+      # Encapsulates subsequent SCR calls into a chroot.
+      #
+      # Raises an exception if something goes wrong.
+      #
+      # @param [#to_s] directory to use as '/' for SCR calls
+      def change_scr_root(directory)
+        # On first call, store the default handler in the stack
+        @scr_handles ||= [Yast::WFM.SCRGetDefault]
```

Comment:
@jreidinger On the other hand, I don't like the idea of automagically calling reset in the after hook because I can imagine situations in which it does the wrong thing. So I think I will.

1) Remove the stack behavior, supporting just one level.
2) Add the check to prevent two consecutive calls to ```change_scr_root``` without a reset.
3) Allow ```change_scr_root``` to be called with a block. So it's up to the user, if we want to explicitly call reset (most likely with a ```change_scr_root``` on before and a ```reset_scr_root``` on after) or if he want to use it in block form (much like File.open in which you have both options).